src/langchain_google_alloydb_pg/utils/pgvector_migrator.py

```diff
@@ -0,0 +1,415 @@
+    params = []
+
+    if use_json_metadata:
+        insert_query = f"INSERT INTO {destination_table} (langchain_id, content, embedding, langchain_metadata) VALUES"
+
+        # Create value clause for the SQL query
+        values_clause = "(:langchain_id, :content, :embedding, :langchain_metadata)"
+        insert_query += values_clause
+
+        # Add parameters
+        for row in data:
+            params.append(
+                {
+                    "langchain_id": row.id,
+                    "content": row.document,
+                    "embedding": row.embedding,
+                    "langchain_metadata": json.dumps(row.cmetadata),
+                }
+            )
+    elif metadata_column_names:
+        insert_query = (
+            f"INSERT INTO {destination_table} (langchain_id, content, embedding"
+        )
+        for column in metadata_column_names:
+            insert_query += ", " + column
+        insert_query += ") VALUES"
+
+        # Create value clause for the SQL query
+        values_clause = (
+            "(:langchain_id, :content, :embedding, "
+            + ", ".join([f":{column}" for column in metadata_column_names])
+            + ")"
+        )
+        insert_query += values_clause
+
+        # Add parameters
+        for row in data:
+            param = {
+                "langchain_id": row.id,
+                "content": row.document,
+                "embedding": row.embedding,
+            }
+            for column in metadata_column_names:
+                # In case the key is not present, add a null value
+                if column in row.cmetadata:
+                    param[column] = row.cmetadata[column]
+                else:
+                    param[column] = None
+            params.append(param)
+
+    # Insert rows
+    async with engine._pool.connect() as conn:
+        await conn.execute(text(insert_query), params)
+        await conn.commit()
+
+
+async def _arun_all_batch_inserts(
+    engine: AlloyDBEngine,
+    data: Sequence[RowMapping],
+    destination_table: str,
+    metadata_column_names: Optional[List[str]],
+    use_json_metadata: Optional[bool] = False,
+    insert_batch_size: int = 1000,
+) -> None:
+    """
+    Insert all data in batches of 1000 insert queries at once.
+
+    Args:
+        data (Sequence[RowMapping]): All the data (to be inserted) belonging to a PGVector collection.
+        destination_table (str): The name of the table to insert the data in.
+        metadata_column_names (str): The metadata columns to be created to keep the data in a row-column format.
+            Optional.
+        use_json_metadata (bool): An option to keep the PGVector metadata as json in the AlloyDB table.
+            Default: False. Optional.
+        insert_batch_size (int): Number of rows to insert at once in the table.
+            Default: 1000.
+    """
+    data_size = len(data)
+    for i in range(data_size // insert_batch_size):
+        await _ainsert_single_batch(
```

Comment:
We don't need to await this instantly. We could use asyncio to await a few coroutines at a time.

Reply:
Done.